packages/markdown/loader.js

```diff
@@ -219,7 +219,9 @@ module.exports = async function demoLoader() {
   );
 
   componentNames.forEach((componentName) => {
-    const moduleID = path.join(this.rootContext, 'src', componentName).replace(/\\/g, '/');
+    const moduleID = path
+      .join(this.rootContext, 'src', componentName.replace(/^docs\/src/, ''))
+      .replace(/\\/g, '/');
```

Comment:
Is this change needed? https://github.com/mui/material-ui/pull/37007/files#r1205292380

```suggestion
    const moduleID = path.join(this.rootContext, 'src', componentName).replace(/\\/g, '/');
``